### DISCUSSION TITLE
Can I import PDF images?

### DISCUSSION THREAD

**Discussion by mcallbee** [keiwando/toonsquid#352] [General]
Can I import PDF images. They are always greyed-out when I try to open them. Thanks

**Comment by Cat-Butler** [comment]
Not in the current version, no.
Strictly speaking, PDFs are a container format, not an image format - they can contain raster (often JPEG) and vector images, text, mathematical formulae using special characters, even fonts and various file attachments.

**Comment by mcallbee** [comment]
Thanks for clarifying. Can I ask you a question here...or does everything
have to be posted to Github?

I changed my images to PNGs (black and white line drawings) and they
imported great. I wanted to color fill a man's tee shirt and it said that
it could not fill that layer. I created another layer and it filled the
whole frame. How do I begin colorizing/filling my black and white drawings??

**Reply by keiwando** [reply]
Hi! Just as a heads-up: When you reply to such a notification-email from GitHub, the reply is automatically posted publicly as a reply to the discussion thread on GitHub.

Regarding the fill question: The fill tool works on pixel layers and vector layers. You can [rasterize](https://toonsquid.com/handbook/layers/layer_list/#actions) other layer types to turn them into pixel layers and then fill them.

**Reply by keiwando** [reply]
And if you have other questions regarding ToonSquid that you don't want to post publicly, you can e-mail contact@toonsquid.com.

**Comment by mcallbee** [comment]
These images were pngs... But I would have thought those were pixels. But I
can't fill them.

**Reply by keiwando** [reply]
When you import an image into ToonSquid, a new separate animation clip is created that contains the pixels of that image. That's why the image also then shows up in the project's library. The layer that is inserted into the main timeline is a [symbol layer](https://toonsquid.com/handbook/symbols/symbol_layer/) that references the clip with the image. This symbol layer needs to be rasterized before you can fill it. You could alternatively also select and edit the image clip. In there you'll find a pixel layer with the actual pixels of the image and you can use the fill tool directly on this layer.

When you use the [image sequence import option](https://toonsquid.com/handbook/layers/pixel_layer/#import-image-sequence), ToonSquid won't create new animation clips for each image but rather downscale them if necessary to fit into the current timeline resolution and put their contents into pixel layers. If that's fine with you and you don't want the separate symbols, you can just use the image sequence import option to import images.
